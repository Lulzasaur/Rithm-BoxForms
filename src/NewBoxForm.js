import React, { Component } from 'react';

class NewBoxForm extends Component {  // ...
    constructor(props) {
        super(props);
        this.state = {height:0,width:0,color:'white'};
    
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
      }

    handleSubmit(evt) {
        evt.preventDefault();
        this.props.handleNewItem(this.state);
        this.setState({height:0, width:0,color:'white'});
      }
    
    handleChange(evt) {
      this.setState({ [evt.target.name]: evt.target.value });
    }
  
    render() {
      return (
          <form onSubmit={this.handleSubmit}>
  
            <label htmlFor="height">Height:</label>
            <input id="height" name="height"
                   value={this.state.height}
                   onChange={this.handleChange} />
  
            <label htmlFor="width">Width:</label>
            <input id="width" name="width"
                   value={this.state.width}
                   onChange={this.handleChange} />

            <label htmlFor="color">Color:</label>
            <input id="color" name="color"
                   value={this.state.color}
                   onChange={this.handleChange} />       

            <button>Add a new box!</button>
  
          </form>
      );
    }
  } // end

  export default NewBoxForm;

