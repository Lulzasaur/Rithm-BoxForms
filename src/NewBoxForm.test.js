import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewBoxForm from './NewBoxForm';

it('renders without crashing', function(){
    shallow(<NewBoxForm/>)
})

it('matches snapshot', function(){
    let wrapper = shallow(<NewBoxForm/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
})

it('test state has changed when we onchange', function(){
    let wrapper = shallow(<NewBoxForm/>);
    wrapper.setState({height:0,width:0,color:'white'})
    const input = wrapper.find('#height').first()
    input.simulate('change',{
        target:{name: "height", value:9999}
    })
    expect(wrapper.state().height).toEqual(9999);
})