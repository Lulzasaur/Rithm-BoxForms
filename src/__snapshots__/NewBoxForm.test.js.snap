// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value={0}
  />
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value={0}
  />
  <label
    htmlFor="color"
  >
    Color:
  </label>
  <input
    id="color"
    name="color"
    onChange={[Function]}
    value="white"
  />
  <button>
    Add a new box!
  </button>
</form>
`;
